Documentation/addressing-incremental-compilation-warnings.md

```diff
@@ -0,0 +1,17 @@
+Addressing Incremental Compilation Warnings
+===========================================
+
+Incremental compilation is unsafe when compilation relies on tools with potential side effects (tools that can cause data races or modify the state of other projects that have been deemed safe to skip compilation. Or tools that integrate timestamps or guids into the build output).
+
+The presence of such cases will turn off incremental compilation.
+
+Warning codes for project structures that are unsafe:
```

Comment:
That's correct, they are the error codes the tool will print out. I'll update the doc to make it clear.